src/components/AudioButton/useAudio.js

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useState, useRef } from 'react'
+import { Howl } from 'howler'
+
+export default function useAudio({ songUrl, id, currentSong }) {
+  const song = new Howl({
+    src: [`${songUrl}`],
+    format: ['mp3'],
+    preload: true,
+    autoplay: false,
+    volume: 0.2,
+    onend: function () {
+      setIsSongPlaying(false)
+    },
+  })
+
+  const [isSongPlaying, setIsSongPlaying] = useState(false)
+  let songRef = useRef(null)
+
+  useEffect(() => {
+    songRef.current = song
+  }, [])
+
+  useEffect(() => {
+    if (currentSong === id && isSongPlaying) {
+      songRef.current.play()
+      songRef.current.fade(0, 0.2, 500)
+    } else {
+      setIsSongPlaying(false)
+      songRef.current.pause()
+    }
+  }, [currentSong, isSongPlaying])
+
+  return { isSongPlaying, setIsSongPlaying }
```

Comment:
Ok, I am not totally sure about it, but I share my thoughts on it:  :)
I have understood, that the functions of a custom hook have to stay in the respective hook. For example ``setIsSongPlaying`` have to stay in the useAudio hook file, because the outer components do not need to know about what happens. Therefore, instead of passing ``setIsSongPlaying``,  an other function (maybe something like ``changeSongStatus`` or better named :D ) could be passed to the respective components, in order to can be called by them. And in this file this function sets ``seIsSongPlaying``. 

```
function changeSongStatus () {
 setIsSongPlaying(!isSongPlaying)
}
```
Could that work?